packages/router/__tests__/path-resolution-test.ts

```diff
@@ -138,7 +138,7 @@ describe("path resolution", () => {
             path: "*",
           },
         ],
-        "/foo",
+        "/foo/bar",
```

Comment:
Similar to #10983, we had a test here asserting the previously buggy behavior.  A link to "." from a splat should stay on the current location, not route up to the parent